enzy_htp/_interface/modeller_interface.py

```diff
@@ -1,27 +1,179 @@
+            if chain is None:
+                continue
+            
+            if aidx > 0 and (ar.chain != all_residues[aidx-1].chain):
+                current_seq += '/'                                
+                target_seq += '/'                                
+            
+            ar.seq_idx = sidx
+            sidx += 1
+            target_seq += ar.one_letter()
+            if ar.missing: 
+                current_seq += '-' 
+            else:
+                if not first_existing:
+                    first_existing = ar
+                last_existing = ar
+                current_seq += ar.one_letter()
+        
+        sp = PDBParser()
+
+        current_seq += '*'
+        target_seq += '*'
+
+        align_file:str = f"{work_dir}/alignment.ali"
+        temp_file:str = f"{work_dir}/modeller_temp.pdb"
+
+        sp.save_structure( temp_file, stru )
+
+        lines:List[str] = [
+            ">P1;modeller_temp",
+           f"structureX:modeller_temp:{first_existing.idx}:{first_existing.chain}:{last_existing.idx}:{last_existing.chain}:undefined:undefined:-1.00:-1.00",
+           f"{current_seq}",
+            ">P1;modeller_fill",
+            "sequence:::::::::",
+           f"{target_seq}"
+        ]
+
+        fs.write_lines(align_file, lines)
+        #TODO(CJ): probably need to check for loops that are too long
+        log.none()
+        env = Environ()
+
+        start_dir:str=f"{Path(os.getcwd()).absolute()}"
+
+        os.chdir(work_dir)
+        
+        # directories for input atom files
+        env.io.atom_files_directory = ['.', '../atom_files']
+        
+        a = LoopModel(env, alnfile = 'alignment.ali',
+                      knowns = 'modeller_temp', sequence = 'modeller_fill')
+        a.starting_model= 1
+        a.ending_model  = 1
+        
+        a.loop.starting_model = 1
+        a.loop.ending_model   = 2
+        a.loop.md_level       = refine.fast
+        
+        a.make()
+       
+        fs.safe_mv("modeller_fill.B99990001.pdb", "modeller_fill.pdb")
+
+        for tk in """modeller_fill.BL00010001.pdb
+        modeller_fill.B99990001.pdb
+        modeller_fill.BL00020001.pdb
+        modeller_fill.D00000001
+        modeller_fill.DL00010001
+        modeller_fill.DL00020001
+        modeller_fill.IL00000001.pdb
+        modeller_fill.ini
+        modeller_fill.lrsr
+        modeller_fill.rsr
+        modeller_fill.sch
+        modeller_fill.V99990001""".split():
+            fs.safe_rm(tk)
+   
+        os.chdir(start_dir)
+
+        session = self.parent().pymol.new_session() 
+        self.parent().pymol.general_cmd(session, [
+            ('load', f"{work_dir}/modeller_fill.pdb"),
+            ("load", f"{work_dir}/modeller_temp.pdb"),
+            ('align', 'modeller_fill', 'modeller_temp'),
+            ('save', f"{work_dir}/modeller_fill.pdb", 'modeller_fill')
+        ])
+
+        filled_stru = sp.get_structure(f"{work_dir}/modeller_fill.pdb")
+
+        if not inplace:
```

Comment:
inplace not defined?

Reply:
Left that in the code by accident. Now removed.